### PR TITLE
add service file for pigeond

### PR DESCRIPTION
**Pull request by aditya-manit**
# Related Github tickets

- Solves no issues

# Background

Add service file for pigeond

# Testing completed

- [ ] Working on running pigeond using this service file, will update in comment in a while

### CODE REVIEW

**Comment by aditya-manit** [comment]
Do not merge yet please, working on testing it. Will update here once done

**Review by taariq — CHANGES_REQUESTED** [review]
@aditya-manit Will you update without the sudo on the commands? We are tending to lean against sudo in the system commands themselves.

File: README.md [hunk 1]
```diff
@@ -109,8 +109,42 @@ export ETH_SIGNING_KEY=<Your ETH SIGNING KEY>
 pigeon start
 ```
 
-- Remember to run pigeon as a systemd Service! If you have a good systemd service implementation for Ubuntu, please make a PR on this README and we will add it.
-
+#### Using systemd service
+
+```bash
```
Comment (measure-fi) [inline README.md:114]:
````
```shell
````

File: README.md [hunk 2]
```diff
@@ -109,8 +109,42 @@ export ETH_SIGNING_KEY=<Your ETH SIGNING KEY>
...
+# create service file
+sudo tee <<EOF >/dev/null /etc/systemd/system/pigeond.service
```
Comment (measure-fi) [inline README.md:116]: We don't typically include sudo in setup commands.

```
cat <<EOT >/etc/systemd/system/pigeond.service
```

File: README.md [hunk 3]
```diff
@@ -109,8 +109,42 @@ export ETH_SIGNING_KEY=<Your ETH SIGNING KEY>
...
+WantedBy=multi-user.target
+EOF
+
+# verify the contents
+cat /etc/systemd/system/pigeond.service
+
+# enable
+systemctl enable pigeond
+# reload daemon
+systemctl daemon-reload
```
Comment (measure-fi) [inline README.md:142]: This is required after modifying something in /etc/systemd, I don't think it's required after creating the file. I would delete this whole block and just include

```
service pigeond start
```

(maybe a line after about checking status or logs)

**Comment by measure-fi** [comment]
@aditya-manit You don't need to make those changes, I can do a followup PR with my particulars. Overall readme changes look probably gtg.

File: README.md (hunk 1 above)
Comment (aditya-manit) [reply README.md]: you mean `shell script` ? I don't readme  highlight just with shell

File: README.md [hunk 4]
```diff
@@ -109,8 +109,42 @@ export ETH_SIGNING_KEY=<Your ETH SIGNING KEY>
...
+
+```bash
+# create service file
+sudo tee <<EOF >/dev/null /etc/systemd/system/pigeond.service
+[Unit]
+Description=Pigeon daemon
+After=network-online.target
+
+[Service]
+Environment="PALOMA_KEYRING_PASS=<your Paloma key password>"
```
[the remark this answers is not in the document]
Comment (aditya-manit) [reply README.md]: There is another way to read env from a file which is more secure, I will that tomorrow sometime

[the remark this answers is not in the document]
Comment (aditya-manit) [reply README.md]: Added the variables

File: README.md [hunk 5]
```diff
@@ -109,8 +109,42 @@ export ETH_SIGNING_KEY=<Your ETH SIGNING KEY>
...
+Restart=on-failure
+RestartSec=3
+LimitNOFILE=4096
+
+[Install]
+WantedBy=multi-user.target
+EOF
+
+# verify the contents
+cat /etc/systemd/system/pigeond.service
```
[the remark this answers is not in the document]
Comment (aditya-manit) [reply README.md]: Done

File: README.md (hunk 3 above)
Comment (aditya-manit) [reply README.md]: Done

**Comment by aditya-manit** [comment]
Done with the changes

**Review by measure-fi — DISMISSED** [review]

**Review by taariq — APPROVED** [review]

File: README.md [hunk 6]
```diff
@@ -109,8 +109,34 @@ export ETH_SIGNING_KEY=<Your ETH SIGNING KEY>
...
+Create service file
+
+```shell script
+tee <<EOF >/dev/null /etc/systemd/system/pigeond.service
+[Unit]
+Description=Pigeon daemon
+After=network-online.target
+
+[Service]
+Environment="PALOMA_KEYRING_PASS=$PALOMA_KEYRING_PASS"
```
Comment (Vizualni) [inline README.md:122]: Does this assume that the person running this command has $PALOMA_KEYRING_PASS and other env variables set when running this command? (@aditya-manit )

Comment (aditya-manit) [reply README.md:122]: Yes, we are setting it here before running that command. Line 103 to line 110

File: README.md [hunk 7]
```diff
@@ -109,8 +109,34 @@ export ETH_SIGNING_KEY=<Your ETH SIGNING KEY>
 pigeon start
 ```
 
-- Remember to run pigeon as a systemd Service! If you have a good systemd service implementation for Ubuntu, please make a PR on this README and we will add it.
-
+#### Using systemd service
+Create service file
+
+```shell script
+tee <<EOF >/dev/null /etc/systemd/system/pigeond.service
```
Comment (Vizualni) [inline README.md:116]: @aditya-manit this should be `pigeon`, not `pigeond`

Comment (Homelander91) [reply README.md:116]: Usually I have seen a `d` suffix in binaries and services too, `d` for daemon I guess. We can remove it if we want to - nothing major.

File: README.md [hunk 8]
```diff
@@ -109,8 +109,34 @@ export ETH_SIGNING_KEY=<Your ETH SIGNING KEY>
...
+[Unit]
+Description=Pigeon daemon
+After=network-online.target
+
+[Service]
+Environment="PALOMA_KEYRING_PASS=$PALOMA_KEYRING_PASS"
+Environment="ETH_RPC_URL=$ETH_RPC_URL"
+Environment="ETH_PASSWORD=$ETH_PASSWORD"
+Environment="ETH_SIGNING_KEY=$ETH_SIGNING_KEY"
```
Comment (Vizualni) [inline README.md:125]: Given the default env variables that we are exporting and based on the config file in this readme, you need to add VALIDATOR environment variable as well.

File: README.md (hunk 7 above)
Comment (Vizualni) [reply README.md:116]: the prefix `d` is a cosmos thing where they add the `d` at the end of a chain executable

**Pull request merged**